src/util/future.rs

```diff
@@ -0,0 +1,61 @@
+use tokio_timer::Deadline;
+
+use futures::Future;
+
+use std::time::Instant;
+
+
+/// An extension trait for `Future` that provides a variety of convenient
+/// combinator functions.
+///
+/// Currently, there only is a [`deadline`] function, but this will increase
+/// over time.
+///
+/// Users are not expected this trait. All types that implement `Future` already
```

Comment:
"Users are not expected to implement this trait" ?

Reply:
yes!